cypress/e2e/functional/teacher_tests/teacher_sort_work_view_spec.js

```diff
@@ -103,7 +103,82 @@ describe('SortWorkView Tests', () => {
     sortWork.getFocusDocument().should("be.visible");
   });
 
-  // TODO: Reinstate the tests below when all metadata documents have the new fields and are updated in real time.
+  it("should open Sort Work tab and test secondary sort functionality", () => {
+    beforeTest(queryParams1);
+
+    cy.get(".section-header-arrow").click({multiple: true}); // Open the sections
+    cy.get("[data-testid=section-sub-header]").should("not.exist");
+    cy.get("[data-testid=doc-group]").should("not.exist");
+    cy.get("[data-testid=doc-group-label]").should("not.exist");
+    cy.get("[data-testid=doc-group-list]").should("not.exist");
+
+    // Switching from "Show for" from Problem to Investigation should switch the list of
+    // documents from the larger thumbnail view to the smaller "simple" view and arrange the
+    // document list items in rows that are potentially scrollable.
+    sortWork.getShowForMenu().click();
+    sortWork.getShowForInvestigationOption().click();
+    cy.get("[data-testid=section-sub-header]").should("not.exist");
+    cy.get("[data-testid=doc-group]").should("exist");
+    // There should be one doc group per section-document-list. There is no
+    // label for the doc group.
+    cy.get("[data-testid=section-document-list]").each($el => {
+      cy.wrap($el).find("[data-testid=doc-group]").should("have.length", 1);
+      cy.wrap($el).find("[data-testid=doc-group-label]").should("not.exist");
+    });
+    cy.get("[data-testid=doc-group-list]").invoke("prop", "scrollLeft").should("be.eq", 0);
+    cy.get("[data-testid=scroll-button-left]").should("exist").and("be.disabled");
+    cy.get("[data-testid=scroll-button-right]").should("exist").and("not.be.disabled");
+    cy.get("[data-testid=scroll-button-right]").click();
+    cy.wait(500);
```

Comment:
I've added comments for the waits. The waits are there to ensure the scroll animation has time to complete before the updated UI state is checked. While I think it's possible for the tests to pass without waiting (they do when I run them in my local dev environment when I remove the waits) in the past I've found that similar tests often fail when run in the CI environment unless you include a short wait. I automatically included them here for that reason. I haven't tried seeing if the checks will pass in CI without waits. Do you think they're likely unnecessary?

Reply:
Generally Cypress expects that things may take time, and will automatically wait up to the configured timeout for conditions to become true. That said, there are some times that a wait is needed.  I do think it would be worth trying first before assuming that they are. A few seconds here and there does add up given the number of tests we have.

Reply:
OK. Looks like it's passing here without the waits. So that's good 👍